src/vs/platform/quickinput/browser/quickPickPin.ts

```diff
@@ -77,13 +77,15 @@ function updateButtons(item: QuickPickItem, removePin: boolean): void {
 	if (item.type === 'separator') {
 		return;
 	}
+
 	// remove button classes before adding the new one
-	item.buttons = item.buttons ? item.buttons?.filter(button => button.iconClass && !buttonClasses.includes(button.iconClass)) : [];
-	item.buttons.unshift({
+	const newButtons = item.buttons ? item.buttons?.filter(button => button.iconClass && !buttonClasses.includes(button.iconClass)) : [];
```

Comment:
If `item.buttons` is defined, can't it be `item.buttons.filter` instead of `item.buttons?.filter`?